include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -2261,146 +2261,16 @@ namespace internal
    * currently supports $0\leq fe\_degree \leq 9$ and $degree+1\leq
    * n\_q\_points\_1d\leq fe\_degree+2$.
    */
-  template <int dim, typename Number>
-  struct FEEvaluationImplEvaluateSelector
+  template <int dim, typename Number, bool do_integrate>
+  struct FEEvaluationImplSelector
   {
-    template <int fe_degree, int n_q_points_1d>
+    template <int fe_degree, int n_q_points_1d, typename VNumber>
```

Comment:
What is `VNumber` and why is it needed? This should be documented; if needed, I suggest to use a more common naming scheme, such as `Number2` or `OtherNumber`.